src/lib/rest/restReply.cpp

```diff
@@ -117,6 +117,26 @@ void restReply(ConnectionInfo* ciP, const std::string& answer)
     }
   }
 
+  if (strlen(restCORSEnabled) > 0)
+  {
+    // If any origin is allowed, the header is sent always with "any" as value
+    if (strcmp(restCORSEnabled, "__ALL") == 0)
+    {
+      MHD_add_response_header(response, "Access-Control-Allow-Origin", "*");
+    }
+    // If a specific origin is allowed, the header is only sent if the origins match
+    else if (strcmp(ciP->httpHeaders.origin.c_str(), restCORSEnabled) == 0)
+    {
+      MHD_add_response_header(response, "Access-Control-Allow-Origin", restCORSEnabled);
+    }
+
+    if (ciP->verb == OPTIONS)
+    {
+      MHD_add_response_header(response, "Access-Control-Allow-Headers", "User-Agent, Fiware-Service, Fiware-Servicepath, Ngsiv2-AttrsFormat, Fiware-Correlator, X-Forwarded-For, X-Real-IP, X-Auth-Token");
```

Comment:
Maybe this is a naïve question as I'm not an expert in CORS but :)... why this particular election of headers?

Reply:
: ) I took a look at the `httpHeaderGet` method of `rest.cpp` and among the headers that are defined there, those are the headers that would trigger a preflight request according to the CORS spec (details [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)).

Reply:
Thanks for the article link! (I think that one could be used in cli.md, as I mention in other comment).

However, it is pretty long and I'd take some time to read it. In the meanwhile, could you point me to the place (section, etc.) in which the criteria to include or not include a given header in Access-Control-Allow-Headers is described, please?

Reply:
Sure [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#Preflighted_requests), you can find the details on headers for the preflighted requests. 

Since the text itself includes additional links, it becomes a long read eventually, sorry :)

Reply:
After looking to the document, the only criteria that I have found (tell me if I'm wrong and it is other, please) is:

> The Access-Control-Allow-Headers header is used in response to a preflight request to indicate which HTTP headers can be used when making the actual request.

So, my original question can be reformulated in the following way: are User-Agent, Fiware-Service, Fiware-Servicepath, Ngsiv2-AttrsFormat, Fiware-Correlator, X-Forwarded-For, X-Real-IP and X-Auth-Token the *only* headers that should be allowed?

Probably yes, but I'd like to have your confirmation on this :)

Reply:
The rest of the headers we process are: 
* Host, Expect, Connection, Content-Length, Origin. All of which are Forbidden header names all of which should not be in the Access-Control-Request-Headers in the first place.
* Accept, Content-Type. These two are CORS-safelisted request headers and they are not required to be checked. 

Thanks for your questsion @fgalan, it made me realize that I skipped User-Agent and also, Content-Type is only listed as CORS-safe for a subset of its values. So I will add both of them to the list.

Having said that, for the next PR, I will check what we have in the Access-Control-Request-Headers and set our Access-Control-Allow-Headers accordingly in order not to create an unnecessarily long response.

More info [here](https://www.w3.org/TR/cors/).

Reply:
Fixed in 543c64b